bench/randindex_bench.d

```diff
@@ -0,0 +1,392 @@
+        enum outer_loop_iterations = count / (modulus_max - modulus_min);
+        enum warmup_outer_loop_iterations = min(outer_loop_iterations / 2, 2_000_000u);
+        ulong s = 0;
+
+        StopWatch sw;
+        sw.start;
+        foreach(_; 0 .. warmup_outer_loop_iterations) //boost CPU
+        {
+            foreach (m; modulus_min .. modulus_max)
+            {
+                s += gen.randIndexV1!uint(m);
+                s += gen.randIndexV2!uint(m);
+            }
+        }
+        sw.stop;
+        sw.reset;
+        gen.__ctor(seed);
+        s = 0;
+        sw.start;
+        foreach(_; 0..outer_loop_iterations)
+        {
+            uint s1 = 0;
+            foreach (m; modulus_min .. modulus_max)
+                s1 += gen.randIndexV1!uint(m);
+            s += s1;
+        }
+        sw.stop;
+        writefln("randIndexV1: %s * 10 ^^ 8 calls/s; sum = %d", double(count) / sw.peek.msecs / 100_000, s);
+        sw.start;
+        foreach(_; 0 .. warmup_outer_loop_iterations) //boost CPU
+        {
+            foreach (m; modulus_min ..modulus_max)
+            {
+                s += gen.randIndexV1!uint(m);
+                s += gen.randIndexV2!uint(m);
+            }
+        }
+        sw.stop;
+        sw.reset;
+        gen.__ctor(seed);
+        s = 0;
+        sw.start;
+        foreach(_; 0..outer_loop_iterations)
+        {
+            uint s1 = 0;
+            foreach (m; modulus_min .. modulus_max)
+                s1 += gen.randIndexV2!uint(m);
+            s += s1;
+        }
+        sw.stop;
+        writefln("randIndexV2: %s * 10 ^^ 8 calls/s; sum = %d", double(count) / sw.peek.msecs / 100_000, s);
+        sw.start;
+        foreach(_; 0 .. warmup_outer_loop_iterations) //boost CPU
+        {
+            foreach (m; modulus_min ..modulus_max)
+            {
+                s += gen.randIndexV1!uint(m);
+                s += gen.randIndex!uint(m);
+            }
+        }
+        sw.stop;
+        sw.reset;
+        gen.__ctor(seed);
+        s = 0;
+        sw.start;
+        foreach(_; 0..outer_loop_iterations)
+        {
+            uint s1 = 0;
+            foreach (m; modulus_min .. modulus_max)
+                s1 += gen.randIndex!uint(m);
+            s += s1;
+        }
+        sw.stop;
+        writefln("new mir.random.randIndex (potential inlining shenanigans): %s * 10 ^^ 8 calls/s; sum = %d", double(count) / sw.peek.msecs / 100_000, s);
+    }
+
+    {
+        //Uniform distribution check.
+        static struct Counter
+        {
+            @nogc nothrow pure @safe:
+            enum bool isRandomEngine = true;
+            enum uint max = uint.max;
+            uint state;
+            @disable this();
+            @disable this(this);
+            this(uint state) { this.state = state; }
+            uint opCall() { return state++; }
+        }
+        import mir.random.engine;
+        enum uint nbuckets = uint(1u << 16);
```

Comment:
The problem with this number is that `uint.max  % nbuckets  == 0`. So the while loop is never checked.